x/reports/types/genesis_test.go

```diff
@@ -0,0 +1,210 @@
+package types_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/desmos-labs/desmos/v3/x/reports/types"
+)
+
+func TestValidateGenesis(t *testing.T) {
+	testCases := []struct {
+		name      string
+		data      *types.GenesisState
+		shouldErr bool
+	}{
+		{
+			name: "duplicated subspaces data returns error",
+			data: types.NewGenesisState([]types.SubspaceDataEntry{
+				types.NewSubspacesDataEntry(1, 1, 1),
+				types.NewSubspacesDataEntry(1, 1, 1),
+			}, nil, nil, types.DefaultParams()),
+			shouldErr: true,
+		},
+		{
+			name: "invalid subspaces data returns error",
+			data: types.NewGenesisState([]types.SubspaceDataEntry{
+				types.NewSubspacesDataEntry(0, 1, 1),
+			}, nil, nil, types.DefaultParams()),
+			shouldErr: true,
+		},
+		{
+			name: "duplicated reason returns error",
+			data: types.NewGenesisState(nil, []types.Reason{
+				types.NewReason(1, 1, "Spam", ""),
+				types.NewReason(1, 1, "Spam", ""),
+			}, nil, types.DefaultParams()),
+			shouldErr: true,
+		},
+		{
```

Comment:
Missing invalid reason case

Reply:
There's no reason to add this test here since it' already present inside the `TestReport_Validate` method. Here we just need to make sure that an invalid report (for whatever reason it might be) can be properly identified and fails the entire genesis validation